src/main/java/org/elasticsearch/index/query/TermQueryParser.java

```diff
@@ -80,27 +74,19 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
             }
             parser.nextToken();
         } else {
-            value = parser.text();
+            value = parser.objectText();
```

Comment:
Thinking about it, I tend to want to revert this change since I'm not sure about it's implications. The problem then is that `Object value` can be of different type in original query and after `fromXContent()`, but in the end it only matters that `toQuery()` produces the same result, and there every Object is converted to lucene BytesRef. Thats why I tend to use that conversion also in the `TermQueryBuilder#equals()`. Will push that change to show what I mean.